### DISCUSSION TITLE
data lineouts

### DISCUSSION THREAD

**Discussion by amylazicki** [HEXRD/hexrdgui#1433] [Ideas]
I am very excited to see that the TARDIS rind is now available in the LLNL import tool as hopefully the direct beam spot and wire shadows will provide an additional constraint on the calibration when the structure is unknown.  Also sometimes if there is texture one of the diffraction peaks shows up most strongly on the rind.  Usually though, there are too many background features on the rind for it to make sense to include it in the data lineout and I suggest you provide the option of excluding that plate (or any of the plates) from the lineout.  Maybe that exists already and I just didn't find it.  I also would find it useful to be able to make lineouts over user-selected regions of the data.  I would like to be able to place a cursor on the lineout and read out the 2-theta position.  I also suggest that you don't default to scaling the lineout together with the color map of the image plates.  Although the log, sqrt etc. scaling of the image plates almost always improves the visibility of the lines on the image plates, it almost always makes the lineouts look weird and amplifies the noise.  At least, if you are going to scale the lineouts together with the image plate color scale, change the y-axis units to make it clear that it's not a linear scale.  Also, could you provide a button to scale the lineout so that the mimimum and maximum  of the data are spanning the full range on the y-axis?  I had thought that the 'reset original view' button with the home icon would do this, but it doesn't appear to, and I am finding after doing a lot of rescaling and zooming in and out that I'm in a state where all but the 'none'(linear) scaling are completely off the scale and I have to use the pan button to try to get the zoom correct.  Thank you!!

**Comment by psavery** [comment]
> I am very excited to see that the TARDIS rind is now available in the LLNL import tool

Thanks Amy! The rind is still a work in progress (note the warning in the "Messages" about cylindrical detectors when you use it). You can currently load it and see it, but there are still some issues using it with certain parts of the GUI (for instance: the yellow detector border lines drawn for it are not correct, and if you apply distortions to powder overlays, they aren't displayed on the rind. There may be some other issues as well that we need to work on).

> as hopefully the direct beam spot and wire shadows will provide an additional constraint on the calibration when the structure is unknown. Also sometimes if there is texture one of the diffraction peaks shows up most strongly on the rind.

Good to know! We should look into adding these constraints.

> Usually though, there are too many background features on the rind for it to make sense to include it in the data lineout and I suggest you provide the option of excluding that plate (or any of the plates) from the lineout. Maybe that exists already and I just didn't find it.

This does not exist. I just added issue #1434 for this!

> I also would find it useful to be able to make lineouts over user-selected regions of the data.

I just added #1435 for this!

> I would like to be able to place a cursor on the lineout and read out the 2-theta position.

If you place your cursor over the lineout, note that in the bottom left corner of the application, there is an `x = ...`. This should show you the two theta value for where your mouse is! That is called the "status bar", and the status bar also shows extra information when you hover your mouse over the main plot, such as hkl, d-spacing, the value, etc.

> I also suggest that you don't default to scaling the lineout together with the color map of the image plates. Although the log, sqrt etc. scaling of the image plates almost always improves the visibility of the lines on the image plates, it almost always makes the lineouts look weird and amplifies the noise. At least, if you are going to scale the lineouts together with the image plate color scale, change the y-axis units to make it clear that it's not a linear scale.

I think we originally didn't scale the lineout, but it was suggested by Joel (see #557, for example). But I just added issue #1437 for this!

> Also, could you provide a button to scale the lineout so that the mimimum and maximum of the data are spanning the full range on the y-axis? I had thought that the 'reset original view' button with the home icon would do this, but it doesn't appear to, and I am finding after doing a lot of rescaling and zooming in and out that I'm in a state where all but the 'none'(linear) scaling are completely off the scale and I have to use the pan button to try to get the zoom correct.

I just added issue #1436 for this. We will look into this!
